torch/distributed/tensor/_sharding_prop.py

```diff
@@ -353,6 +353,8 @@ def propagate_op_sharding_non_cached(self, op_schema: OpSchema) -> OutputShardin
                                 for _ in range(len(op_schema.op._schema.returns))
                             ]
                         )
+                elif op_schema.return_type_list_tensor_like():
```

Comment:
it should at least be folded into the next `elif` branch instead of making it separate